metrics-caffeine/src/main/java/com/codahale/metrics/caffeine/MetricsStatsCounter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 Ben Manes. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.codahale.metrics.caffeine;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import com.github.benmanes.caffeine.cache.stats.StatsCounter;
+
+/**
+ * A {@link StatsCounter} instrumented with Dropwizard Metrics.
+ *
+ * @author ben.manes@gmail.com (Ben Manes)
+ */
+public final class MetricsStatsCounter implements StatsCounter {
+  private final Counter hitCount;
+  private final Counter missCount;
+  private final Counter loadSuccessCount;
+  private final Counter loadFailureCount;
+  private final Timer totalLoadTime;
+  private final Counter evictionCount;
+  private final Counter evictionWeight;
+
+  /**
+   * Constructs an instance for use by a single cache.
+   *
+   * @param registry the registry of metric instances
+   * @param metricsPrefix the prefix name for the metrics
+   */
+  public MetricsStatsCounter(MetricRegistry registry, String metricsPrefix) {
+    requireNonNull(metricsPrefix);
+    hitCount = registry.counter(metricsPrefix + ".hits");
+    missCount = registry.counter(metricsPrefix + ".misses");
+    totalLoadTime = registry.timer(metricsPrefix + ".loads");
+    loadSuccessCount = registry.counter(metricsPrefix + ".loads-success");
+    loadFailureCount = registry.counter(metricsPrefix + ".loads-failure");
+    evictionCount = registry.counter(metricsPrefix + ".evictions");
+    evictionWeight = registry.counter(metricsPrefix + ".evictions-weight");
```

Comment:
Please use [`MetricRegistry#name(String, String...)`](https://javadoc.io/static/io.dropwizard.metrics/metrics-core/4.1.9/com/codahale/metrics/MetricRegistry.html#name-java.lang.String-java.lang.String...-)) to generate the names.

```suggestion
    hitCount = registry.counter(MetricRegistry.name(metricsPrefix, "hits"));
    missCount = registry.counter(MetricRegistry.name(metricsPrefix, "misses"));
    totalLoadTime = registry.timer(MetricRegistry.name(metricsPrefix, "loads"));
    loadSuccessCount = registry.counter(MetricRegistry.name(metricsPrefix, "loads-success"));
    loadFailureCount = registry.counter(MetricRegistry.name(metricsPrefix, "loads-failure"));
    evictionCount = registry.counter(MetricRegistry.name(metricsPrefix, "evictions"));
    evictionWeight = registry.counter(MetricRegistry.name(metricsPrefix, "evictions-weight"));
```